src/main/java/interface_adapter/navigation/NavPresenter.java

```diff
@@ -0,0 +1,57 @@
+package interface_adapter.navigation;
+
+import interface_adapter.ViewManagerModel;
+import interface_adapter.change_password.LoggedInViewModel;
+import interface_adapter.login.LoginViewModel;
+import interface_adapter.recipe_collection.RecipeCollectionViewModel;
+import interface_adapter.search.SearchViewModel;
+import interface_adapter.signup.SignupViewModel;
+import use_case.navigation.NavOutputBoundary;
+
+public class NavPresenter implements NavOutputBoundary {
+    private final SearchViewModel searchViewModel;
+    private final RecipeCollectionViewModel recipeCollectionViewModel;
+    private final LoggedInViewModel loggedInViewModel;
+    private final LoginViewModel loginViewModel;
+    private final ViewManagerModel viewManagerModel;
+
+    public NavPresenter(ViewManagerModel viewManagerModel,
+                            SearchViewModel searchViewModel,
+                            RecipeCollectionViewModel recipeCollectionViewModel,
+                            LoggedInViewModel loggedInViewModel,
+                            LoginViewModel loginViewModel
+                        ) {
```

Comment:
I think you should move this back one line to be at the end of "loginViewModel".